graph/multi/directed.go

```diff
@@ -0,0 +1,275 @@
+// Copyright ©2014 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package multi
+
+import (
+	"fmt"
+
+	"gonum.org/v1/gonum/graph"
+	"gonum.org/v1/gonum/graph/internal/uid"
+)
+
+var (
+	dg *DirectedGraph
+
+	_ graph.Graph              = dg
```

Comment:
I'm OK with this small additional space cost for the benefit of readability. I should change how simple does this - I prefer to have it in the built package and not just in the test files since tests may not be run.

Reply:
Actually, I `dg` would add to the space of the executable was not part of my rationale for removing it. Rather, I consider it code hygiene to limit the number of global variables. And seeing as this code base is spread across multiple files, for a user not yet accustomed to the code base it may take some time to figure out that `dg` is not used. That was the rationale for removing it entirely, and simply using `var _ graph.Graph = (*DirectedGraph)(nil)` as the `_` identifier can never be used.

Reply:
The var is unexported and in a var block. This is a reasonably commonly used convention.

Reply:
the "trick" @mewmew refers to is also a very reasonable and commonly used convention :)
(one could also add a little comment explaining what it does and why)

Reply:
It's the same 'trick', the only difference is the dialect. We have used the dialect here in other packages, largely because of the verbosity of doing it the other way. The existence of an apparently no-op var block like this reads as clearly as a comment; I really don't think a change is necessary here.